api/index.ts

```diff
@@ -1,25 +1,32 @@
-require('dotenv').config();
-const express = require('express');
-const cors = require('cors');
-const { OpenAI } = require('openai');
-const http = require('http');
-const { Server } = require('socket.io');
+import {ClientToServerEvents, InterServerEvents, ServerToClientEvents, SocketData} from "apiTypes";
 
+import { config as dotenvConfig } from "dotenv";
+import { Server } from "socket.io";
+import express from "express";
+import cors from "cors";
+import { OpenAI } from "openai";
+import http from "http";
+
+dotenvConfig()
 const app = express();
 const PORT = 5000;
 
 // Middleware
-app.use(cors({ origin: 'http://localhost:5173', credentials: true }));
-app.use(cors({ origin: 'http://localhost:5174', credentials: true }));
+app.use(cors({ origin: ['http://localhost:5173', 'http://localhost:5174'], credentials: true }));
```

Comment:
We actually don't need port 5174 here if you want to remove that I'd appreciate it. It was an accident check-in on my end (it's what vite goes to if you run a second instance without closing the first).